packages/dds/tree/src/simple-tree/core/context.ts

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { anchorSlot, type TreeNodeSchemaIdentifier } from "../../core/index.js";
+import type {
+	FlexTreeContext,
+	FlexTreeHydratedContext,
+} from "../../feature-libraries/index.js";
+import { brand } from "../../util/index.js";
+import type { TreeNodeSchema } from "./treeNodeSchema.js";
+import { walkAllowedTypes } from "./walkSchema.js";
+
+/**
+ * Creating multiple simple tree contexts for the same branch, and thus with the same underlying AnchorSet does not work due to how TreeNode caching works.
+ * This slot is used to detect if one already exists and error if creating a second.
+ */
+export const SimpleContextSlot = anchorSlot<HydratedContext>();
```

Comment:
Is this pattern the intended long-term solution? Or a temporary one? I don't have a lot of context into how anchors work, so it isn't clear to me if we're using their slots here for their intended purpose, or if we're hijacking an existing system for a temp solution.

Reply:
I think it's fine - this is more or less what anchor slots are intended for AFAIK.